tests/test_types_union.py

```diff
@@ -270,7 +270,7 @@ def test_union_definition_nested(cs: cstruct, compiled: bool) -> None:
     assert isinstance(obj.c.a, UnionProxy)
     assert len(obj.c.a) == 8
     assert bytes(obj.c.a) == b"holybeef"
-    assert repr(obj.c.a) == "<a a=0x796c6f68 b=0x66656562>"
+    assert repr(obj.c.a) == "<__anonymous_0__ a=0x796c6f68 b=0x66656562>"
```

Comment:
I'm not sure if this is better or worse. I think with this change to the field names and anonymous structure names, we're technically more correct, but it's less legible when using e.g. `dumpstruct`.

Maybe we can patch the class and qual-name when we learn of the field name?

Reply:
I looked at this briefly but let's just keep this behavior for now and look at it later. It's easier for the stub generation. We can make it prettier later.